src/MainWindow.vala

```diff
@@ -155,18 +155,33 @@ public class Tasks.MainWindow : Hdy.ApplicationWindow {
 
             listbox.row_selected.connect ((row) => {
                 if (row != null) {
+                    Tasks.ListView? listview;
+
                     if (row is Tasks.SourceRow) {
                         var source = ((Tasks.SourceRow) row).source;
-                        listview.source = source;
-                        Tasks.Application.settings.set_string ("selected-list", source.uid);
+                        var source_uid = source.dup_uid ();
 
-                        listview.add_view (source, "(contains? 'any' '')");
+                        listview = (Tasks.ListView) listview_stack.get_child_by_name (source_uid);
+                        if (listview == null) {
+                            listview = new Tasks.ListView (source);
+                            listview_stack.add_named (listview, source_uid);
+                            listview.add_view (source, "(contains? 'any' '')");
+                        }
 
+                        listview_stack.set_visible_child_name (source_uid);
+                        Tasks.Application.settings.set_string ("selected-list", source_uid);
                         ((SimpleAction) lookup_action (ACTION_DELETE_SELECTED_LIST)).set_enabled (source.removable);
 
                     } else if (row is Tasks.ScheduledRow) {
-                        listview.source = null;
-                        Tasks.Application.settings.set_string ("selected-list", "scheduled");
+                        var scheduled_uid = "scheduled";
```

Comment:
perhaps this should be a constant since it is referenced in a few other places like saving and restoring the selection

Reply:
Good point! Added the `SCHEDULED_LIST_UID` constant.